e2e/framework/pages/heroes/fragments/heroes-list-fragment.ts

```diff
@@ -2,55 +2,45 @@ import { Page } from "playwright";
 import { BaseFragment } from "../../../../libs/base.fragment";
 import { waits } from "../../../../libs/helpers/waits";
 import { Hero } from "../../../models/hero";
-import { DetailsPage } from "../../details";
+import { DetailsPage } from "../../details/details-page";
 
 export class HeroesList extends BaseFragment {
-    private heroesListLocator: string = "//ul[@class='heroes']/li";
-    private heroesIdNameLocator: string = `${this.heroesListLocator}/a`;
+    private heroesListSelector: string = "//ul[@class='heroes']/li";
+    private heroesIdNameSelector: string = `${this.heroesListSelector}/a`;
 
     constructor(page: Page) {
         super(page);
     }
 
     async getHeroesIdNamePairs(): Promise<Hero[]> {
-        await (await waits(this.page).waitVisibility(this.heroesIdNameLocator))
-        let allLinesOfText = await this.page.locator(this.heroesIdNameLocator).allInnerTexts();
+        await waits(this.page).waitVisibility(this.heroesIdNameSelector)
+        let allLinesOfText = await this.page.locator(this.heroesIdNameSelector).allInnerTexts();
         if (allLinesOfText.length === 0) {
             throw new Error('Heroes count should not be null');
         }
         return allLinesOfText.map(val => this.convertStringToHero(val)); 
     }
 
     async selectHeroById(heroId: number) : Promise<DetailsPage> {
-        await this.page.click(`//ul/li/a[span[contains(text(), '${heroId}')]]`);
+        await this.page.locator(`text=${heroId}`).click();
         return new DetailsPage(this.page);
     };
 
-    //TODO: Need to override. IT will be nice ti get only first element. 
-    async getFirstHero() {
-        await (await waits(this.page).waitVisibility(this.heroesIdNameLocator))
-        let allLinesOfText = await this.page.locator(this.heroesIdNameLocator).allInnerTexts();
+    async getFirstHero(): Promise<Hero> {
+        await waits(this.page).waitVisibility(this.heroesIdNameSelector);
 
-        if (allLinesOfText.length === 0) {
-            throw new Error('Heroes count should not be null');
-        }
-        else {
-            return this.convertStringToHero(allLinesOfText[0]);
-        }
+        let firstElement = await this.page.$(this.heroesIdNameSelector);
+        const text = await firstElement?.textContent() as unknown as string;
```

Comment:
Seems unsafe, if you want to have `string` in call cases can do `await firstElement?.textContent() ?? ''`

Reply:
Ok. Thanks. Fixed.